celery/backends/azureblockblob.py

```diff
@@ -0,0 +1,160 @@
+except ImportError:  # pragma: no cover
+    azurestorage = BlockBlobService = ExponentialRetry = \
+        AzureMissingResourceHttpError = None  # noqa
+
+__all__ = ("AzureBlockBlobBackend",)
+
+
+CONF_CONTAINER_NAME = "azureblockblob_container_name"
+CONF_RETRY_INITIAL_BACKOFF_SEC = "azureblockblob_retry_initial_backoff_sec"
+CONF_RETRY_INCREMENT_BASE = "azureblockblob_retry_increment_base"
+CONF_RETRY_MAX_ATTEMPTS = "azureblockblob_retry_max_attempts"
+
+DEFAULT_CONTAINER_NAME = "celery"
+DEFAULT_RETRY_INITIAL_BACKOFF_SEC = 2
+DEFAULT_RETRY_INCREMENT_BASE = 2
+DEFAULT_RETRY_MAX_ATTEMPTS = 3
+
+URL_RE = re.compile(r"azureblockblob://(?P<connection_string>.+)")
+
+LOGGER = get_logger(__name__)
+
+
+class AzureBlockBlobBackend(KeyValueStoreBackend):
+    """Azure Storage Block Blob backend for Celery."""
+
+    def __init__(self,
+                 url=None,
+                 container_name=None,
+                 retry_initial_backoff_sec=None,
+                 retry_increment_base=None,
+                 retry_max_attempts=None,
+                 *args,
+                 **kwargs):
+        super(AzureBlockBlobBackend, self).__init__(*args, **kwargs)
+
+        if azurestorage is None:
+            raise ImproperlyConfigured(
+                "You need to install the azure-storage library to use the "
+                "AzureBlockBlob backend")
+
+        conf = self.app.conf
+
+        self._connection_string = self._parse_url(url)
+
+        self._container_name = (
+            container_name or
+            conf.get(CONF_CONTAINER_NAME) or
+            DEFAULT_CONTAINER_NAME)
+
+        self._retry_initial_backoff_sec = (
+            retry_initial_backoff_sec or
+            conf.get(CONF_RETRY_INITIAL_BACKOFF_SEC) or
+            DEFAULT_RETRY_INITIAL_BACKOFF_SEC)
+
+        self._retry_increment_base = (
+            retry_increment_base or
+            conf.get(CONF_RETRY_INCREMENT_BASE) or
+            DEFAULT_RETRY_INCREMENT_BASE)
+
+        self._retry_max_attempts = (
+            retry_max_attempts or
+            conf.get(CONF_RETRY_MAX_ATTEMPTS) or
+            DEFAULT_RETRY_MAX_ATTEMPTS)
+
+    @classmethod
+    def _parse_url(cls, url):
+        match = URL_RE.match(url)
+        if not match:
+            raise ImproperlyConfigured("Invalid URL")
+
+        return match.group("connection_string")
+
+    @cached_property
+    def _client(self):
+        """Return the Azure Storage Block Blob service.
+
+        If this is the first call to the property, the client is created and
+        the container is created if it doesn't yet exist.
+
+        """
+        client = BlockBlobService(connection_string=self._connection_string)
+
+        created = client.create_container(
+            container_name=self._container_name, fail_on_exist=False)
+
+        if created:
+            LOGGER.info("Created container %s", self._container_name)
+
+        client.retry = ExponentialRetry(
```

Comment:
I was expecting this class to be using constructor-based dependency injection, but [azure-storage sample](https://github.com/Azure/azure-storage-python/blob/19d845d58551bd36a04c46628dc42089ee269362/samples/advanced/client.py#L42) also uses the same pattern.